### PR TITLE
Add message_thread_id for Telegram contact points

### PR DESCRIPTION
**Pull request by th0th**
See https://github.com/prometheus/alertmanager/pull/3638.

### CODE REVIEW

File: internal/resources/grafana/resource_alerting_contact_point_notifiers.go [hunk 1]
```diff
@@ -1701,6 +1707,7 @@ func (t telegramNotifier) unpack(raw interface{}, name string) gapi.ContactPoint
 
 	unpackNotifierStringField(&json, &settings, "token", "bottoken")
 	unpackNotifierStringField(&json, &settings, "chat_id", "chatid")
+	unpackNotifierStringField(&json, &settings, "message_thread_id", "messagehtreadid")
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_contact_point_notifiers.go]:
```suggestion
	unpackNotifierStringField(&json, &settings, "message_thread_id", "messagethreadid")
```

File: internal/resources/grafana/resource_alerting_contact_point_test.go [hunk 2]
```diff
@@ -264,6 +264,7 @@ func TestAccContactPoint_notifiers(t *testing.T) {
 					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.#", "1"),
 					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.0.token", "token"),
 					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.0.chat_id", "chat-id"),
+					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.0.message_thread_id", "message-thread-id"),
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_contact_point_test.go]: hmm wondering why that passes. That's not the value in the test file

Comment (julienduchesne) [reply internal/resources/grafana/resource_alerting_contact_point_test.go]: Oh wait, it doesn't. Tests hadn't run 😄

File: internal/resources/grafana/resource_alerting_contact_point_notifiers.go (hunk 1 above)
Comment (th0th) [reply internal/resources/grafana/resource_alerting_contact_point_notifiers.go]: Nice catch 😅

**Review by spinillos — APPROVED** [review]

**Comment by th0th** [comment]
I synced my branch with upstream 👍

**Comment by julienduchesne** [comment]
Can't really merge this until it's supported in Grafana unfortunately

**Comment by vanpt1114** [comment]
I think Grafana Alerting has supported the Telegram message_thread_id, I can observe it from the UI, so is this PR good to merge?

**Pull request merged**